lib/pure/strformat.nim

```diff
@@ -525,7 +525,13 @@ proc format*(value: SomeFloat; specifier: string; res: var string) =
 
   var f = formatBiggestFloat(value, fmode, spec.precision)
   if value >= 0.0 and spec.sign != '-':
-    f = spec.sign & f
+    if  value == 0.0:
```

Comment:
To make all of this shorter, it would be `if spec.sign != '-' and (value > 0.0 or (value == 0.0 and 1.0 / value == Inf))`

Reply:
maybe shorter but harder to understand for average programmers like me. and it would be difficult to place the comment about the not self-evident purpose of the `1.0 / value == Inf` check.